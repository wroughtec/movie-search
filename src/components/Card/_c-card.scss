.c-card {
  display: flex;
  width: 100%;

  flex-direction: column;
  align-items: center;
}

.c-card__header {
  font-weight: 700;
}

.c-card__title {
  margin-top: 0;
  margin-bottom: 0;

  font-size: 1.25rem;
}
