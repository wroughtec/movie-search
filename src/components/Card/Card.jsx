import React from 'react';
import { Poster } from '../Poster/Poster';
import './_c-card.scss';
import { trimString } from '../../utils/trimString';

export const Card = ({ title, imageBaseUrl, size, path, overview, releaseDate }) => {
  const trimmedOverview = trimString(overview);
  return (
    <article className="c-card">
      <Poster imageBaseUrl={imageBaseUrl} size={size} path={path} title={title} />
      <div className="c-card__body">
        <header className="c-card__header">
          <h2 className="c-card__title">{title}</h2>
          <small className="c-card__date">{releaseDate}</small>
        </header>
        <p className="c-card__overview">{trimmedOverview}</p>
      </div>
    </article>
  );
};
