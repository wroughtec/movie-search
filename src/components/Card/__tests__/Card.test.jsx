import React from 'react';
import { shallow } from 'enzyme';
import { Card } from 'components/Card/Card';

describe('Card', () => {
  let component, props;

  beforeEach(() => {
    props = {
      title: 'Title',
      imageBaseUrl: 'imageBaseUrl',
      size: 'size',
      path: 'path',
      overview: 'overview',
      releaseDate: 'releaseDate'
    };
    component = shallow(<Card {...props} />);
  });
  test('Renders', () => {
    expect(component).toMatchSnapshot();
  });
});
