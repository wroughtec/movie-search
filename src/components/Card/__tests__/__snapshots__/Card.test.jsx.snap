// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    imageBaseUrl="imageBaseUrl"
    overview="overview"
    path="path"
    releaseDate="releaseDate"
    size="size"
    title="Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Poster
          imageBaseUrl="imageBaseUrl"
          path="path"
          size="size"
          title="Title"
        />,
        <div
          className="c-card__body"
        >
          <header
            className="c-card__header"
          >
            <h2
              className="c-card__title"
            >
              Title
            </h2>
            <small
              className="c-card__date"
            >
              releaseDate
            </small>
          </header>
          <p
            className="c-card__overview"
          >
            overview
          </p>
        </div>,
      ],
      "className": "c-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "imageBaseUrl": "imageBaseUrl",
          "path": "path",
          "size": "size",
          "title": "Title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header
              className="c-card__header"
            >
              <h2
                className="c-card__title"
              >
                Title
              </h2>
              <small
                className="c-card__date"
              >
                releaseDate
              </small>
            </header>,
            <p
              className="c-card__overview"
            >
              overview
            </p>,
          ],
          "className": "c-card__body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="c-card__title"
                >
                  Title
                </h2>,
                <small
                  className="c-card__date"
                >
                  releaseDate
                </small>,
              ],
              "className": "c-card__header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Title",
                  "className": "c-card__title",
                },
                "ref": null,
                "rendered": "Title",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "releaseDate",
                  "className": "c-card__date",
                },
                "ref": null,
                "rendered": "releaseDate",
                "type": "small",
              },
            ],
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "overview",
              "className": "c-card__overview",
            },
            "ref": null,
            "rendered": "overview",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Poster
            imageBaseUrl="imageBaseUrl"
            path="path"
            size="size"
            title="Title"
          />,
          <div
            className="c-card__body"
          >
            <header
              className="c-card__header"
            >
              <h2
                className="c-card__title"
              >
                Title
              </h2>
              <small
                className="c-card__date"
              >
                releaseDate
              </small>
            </header>
            <p
              className="c-card__overview"
            >
              overview
            </p>
          </div>,
        ],
        "className": "c-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "imageBaseUrl": "imageBaseUrl",
            "path": "path",
            "size": "size",
            "title": "Title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <header
                className="c-card__header"
              >
                <h2
                  className="c-card__title"
                >
                  Title
                </h2>
                <small
                  className="c-card__date"
                >
                  releaseDate
                </small>
              </header>,
              <p
                className="c-card__overview"
              >
                overview
              </p>,
            ],
            "className": "c-card__body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2
                    className="c-card__title"
                  >
                    Title
                  </h2>,
                  <small
                    className="c-card__date"
                  >
                    releaseDate
                  </small>,
                ],
                "className": "c-card__header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Title",
                    "className": "c-card__title",
                  },
                  "ref": null,
                  "rendered": "Title",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "releaseDate",
                    "className": "c-card__date",
                  },
                  "ref": null,
                  "rendered": "releaseDate",
                  "type": "small",
                },
              ],
              "type": "header",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "overview",
                "className": "c-card__overview",
              },
              "ref": null,
              "rendered": "overview",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
